### PR TITLE
Added tests for Mixed with 16 and 32 bit data

### LINKED ISSUE #4373: Add tests for 16-bit and 23-bit data in Mixed

### PR DESCRIPTION
**Pull request by fronck**
## What, How & Why?

Added tests to verify `Mixed` binary operations with 16-bit and 32-bit values.

This closes #4373.

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Review by kneth — APPROVED** [review]

File: tests/js/mixed-tests.js [hunk 1]
```diff
@@ -238,20 +240,50 @@ module.exports = {
...
+    const uint16Objects = realm.objects("MixedWithData");
+    returnedData = [...new Uint16Array(uint16Objects[0].value)];
+    TestCase.assertArraysEqual(returnedData, uint16Values);
+
+    realm.write(() => {
+      realm.deleteAll();
+    });
+
+    // test with 32-bit values
+    const uint32Values = [75025, 121393, 121393, 317811, 514229, 832040];
```
Comment (kraenhansen) [inline tests/js/mixed-tests.js:278]:
```suggestion
    const uint32Values = [75025, 121393, 121393, 317811, 514229, 4294967295];
```

File: tests/js/mixed-tests.js [hunk 2]
```diff
@@ -238,20 +240,50 @@ module.exports = {
...
     returnedData = [...new Uint8Array(mixedObjects[0].value)];
-    TestCase.assertArraysEqual(returnedData, [255, 128, 64, 32, 16, 8]);
+    TestCase.assertArraysEqual(returnedData, uint8Values2);
+
+    realm.write(() => {
+      realm.deleteAll();
+    });
+
+    // test with 16-bit values
+    const uint16Values = [1024, 512, 256, 128];
```
Comment (kraenhansen) [inline tests/js/mixed-tests.js:263]:
```suggestion
    const uint16Values = [0, 512, 256, 65535];
```

File: tests/js/mixed-tests.js [hunk 3]
```diff
@@ -227,8 +227,10 @@ module.exports = {
 
   // test Mixed datatype with binary data contents
   testMixedData() {
-    const buffer1 = new Uint8Array([1, 2, 4, 8]).buffer;
-    const buffer2 = new Uint8Array([255, 128, 64, 32, 16, 8]).buffer;
+    const uint8Values1 = [1, 2, 4, 8];
```
Comment (kraenhansen) [inline tests/js/mixed-tests.js:230]:
```suggestion
    const uint8Values1 = [0, 1, 2, 4, 8];
```

**Review by kraenhansen — APPROVED** [review]
Looks about right. I've suggested changes to test the extreme values.

**Comment by fronck** [comment]
> Looks about right. I've suggested changes to test the extreme values.

Suggestions applied :+1:

**Pull request merged**